runtime/Go/antlr/common_token_stream.go

```diff
@@ -331,10 +331,12 @@ func (c *CommonTokenStream) GetTextFromRuleContext(interval RuleContext) string
 
 func (c *CommonTokenStream) GetTextFromInterval(interval *Interval) string {
 	c.lazyInit()
-	c.Fill()
 
 	if interval == nil {
+		c.Fill()
 		interval = NewInterval(0, len(c.tokens)-1)
+	} else {
+		c.Sync(interval.Stop)
```

Comment:
We need to double check whether this is supposed to be `interval.Stop` or `interval.Stop - 1`. `Sync` is documented as being an index and `interval.Stop` is documented as being exclusive. However there is some oddities on whether this may or may not be true.

Reply:
Okay, intervals in Go are seriously broken. We are not going to fix it in this pull request, so if this works it works.

Reply:
When I do `interval.Stop-1` it would return empty text in the case where the token is a single character, so I believe the code is correct only because the `Sync` call is compensating by adding another character to fetch under the covers. The same behavior exists in Java as well.

Reply:
Hmm... Everything was inclusive if I remember correctly...yeah, we should take a look at this in more detail.